components/gfx/font_template.rs

```diff
@@ -128,95 +123,97 @@ impl FontTemplate {
         &self.identifier
     }
 
+    /// Returns a reference to the bytes in this font if they are in memory.
+    /// This function never performs disk I/O.
+    pub fn data_if_in_memory(&self) -> Option<Arc<Vec<u8>>> {
+        self.data.clone()
+    }
+
+    /// Returns a [`NativeFontHandle`] for this font template, if it is local.
+    pub fn native_font_handle(&self) -> Option<NativeFontHandle> {
+        match &self.identifier {
+            FontIdentifier::Local(local_identifier) => local_identifier.native_font_handle(),
+            FontIdentifier::Web(_) => None,
+        }
+    }
+}
+
+pub trait FontTemplateRefMethods {
+    /// Returns a reference to the data in this font. This may be a hugely expensive
+    /// operation (depending on the platform) which performs synchronous disk I/O
+    /// and should never be done lightly.
+    fn data(&self) -> Arc<Vec<u8>>;
+    /// Return true if this is a valid [`FontTemplate`] ie it is possible to construct a [`FontHandle`]
+    /// from its data.
+    fn is_valid(&self) -> bool;
+    fn instantiate(&self) -> Result<(), &'static str>;
```

Comment:
Would it be helpful if `instantiate` also had a docstring? For example, I find it hard to judge how it is different from the lazy loading performed by `data`.

Reply:
Sure. I've added a docstring for this method.